packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,287 @@
+  },
+  {
+    section: 'Last',
+    timeframeText: '6 hours',
+    timeframeLength: 60 * 60 * 6
+  },
+  {
+    section: 'Last',
+    timeframeText: '12 hours',
+    timeframeLength: 60 * 60 * 12
+  },
+  {
+    section: 'Last',
+    timeframeText: '24 hours',
+    timeframeLength: 60 * 60 * 24
+  },
+  {
+    section: 'Last',
+    timeframeText: '7 days',
+    timeframeLength: 60 * 60 * 24 * 7
+  },
+  {
+    section: 'Last',
+    timeframeText: '30 days',
+    timeframeLength: 60 * 60 * 24 * 30
+  },
+  {
+    section: 'Last',
+    timeframeText: '90 days',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Week',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Month',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Quarter',
+    timeframeLength: 60 * 60 * 24 * 90
+  }
+]
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    value: {
+      type: Object,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      allowedTimePeriods,
+      mode: this.mode || 'custom',
+      selectimedTimeRange: this.selectimedTimeRange,
+      selectedTimeframe: this.value || this.allowedTimePeriods[0],
+      modelConfig: {
+        type: 'number'
+      }
+    }
+  },
+
+  computed: {
+    showCalendar () {
+      return (this.mode === 'custom')
+    }
+  },
+
+  watch: {
+    selectimedTimeRange: {
+      handler (newValue, oldValue) {
+        // console.warn('>>>> selectimedTimeRange watcher <<<')
+        // console.warn(newValue)
+      },
+      immediate: true
+    }
+  },
+
+  methods: {
+    /**
+     * TODO: Given a relative time frame, determine `start` and `end` timestamps in UTC
+     * https://bloop.ai/search/how-to-use-date-fns-library-round-to-nearest-minutes-in-javascript
+     * @param {*} timeframe
+     */
+    changeTimeframe (timeframe) {
+      this.selectedTimeframe = timeframe
+    },
+    clearSelection () {
+      this.selectimedTimeRange = ''
+      this.selectedTimeframe = this.allowedTimePeriods[0]
+    },
+    submitTimeFrame () {
+      console.warn('>>> submitTimeFrame')
+      let start = 0
+      let end = 0
+
+      // If calendar currently shown, send start/end values
+      if (this.showCalendar) {
+        start = Math.floor(this.selectimedTimeRange.start / 1000)
+        end = Math.floor(this.selectimedTimeRange.end / 1000)
+      } else {
+        end = getUnixTime(roundToNearestMinutes(Date.now()))
+        start = end - this.selectedTimeframe.timeframeLength
+      }
+
+      console.log({ start, end })
+
+      this.$emit('changed', { start, end })
+    }
+  }
+}
+
+</script>
+
+<style lang="scss">
+@import '~@kongponents/styles/variables';
+
+$margin: .2rem;
+
+.analytics-timepicker {
```

Comment:
The wrapper class (outer-most class on the Kongponent) should follow our `k-*` syntax, so can we change this and it's usage at the top of the component file to what I have here? 
```suggestion
.k-datetime-picker {
```